android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt

```diff
@@ -15,127 +14,89 @@ import io.flutter.plugin.common.MethodChannel.Result
                 sdkName = "posthog-flutter"
-                sdkVersion = postHogVersion
+                sdkVersion = PostHog.VERSION
             }
             PostHogAndroid.setup(applicationContext, config)
 
-        } catch (e: Throwable) {
-            e.localizedMessage?.let { Log.e("PostHog", "initPlugin error: $it") }
+        } catch (e: Exception) {
+            Log.e("PostHog", "initPlugin error: ${e.localizedMessage}")
         }
     }
 
     override fun onMethodCall(call: MethodCall, result: Result) {
-
         when (call.method) {
+            "configure" -> configure(call, result)
+            "identify" -> identify(call, result)
+            "capture" -> capture(call, result)
+            "screen" -> screen(call, result)
+            "alias" -> alias(call, result)
+            "distinctId" -> distinctId(result)
+            "reset" -> reset(result)
+            "disable" -> disable(result)
+            "enable" -> enable(result)
+            "debug" -> debug(call, result)
+            "register" -> register(call, result)
+            "unregister" -> unregister(call, result)
+            "flush" -> flush(result)
+            "isFeatureEnabled" -> isFeatureEnabled(call, result)
+            "reloadFeatureFlags" -> reloadFeatureFlags(result)
+            "group" -> group(call, result)
+            "getFeatureFlag" -> getFeatureFlag(call, result)
+            "getFeatureFlagPayload" -> getFeatureFlagPayload(call, result)
+            else -> result.notImplemented()
+        }
+    }
 
-            "identify" -> {
-                identify(call, result)
-            }
-
-            "capture" -> {
-                capture(call, result)
-            }
-
-            "screen" -> {
-                screen(call, result)
-            }
-
-            "alias" -> {
-                alias(call, result)
-            }
-
-            "distinctId" -> {
-                distinctId(result)
-            }
-
-            "reset" -> {
-                reset(result)
-            }
-
-            "disable" -> {
-                disable(result)
-            }
-
-            "enable" -> {
-                enable(result)
-            }
-
-            "isFeatureEnabled" -> {
-                isFeatureEnabled(call, result)
-            }
-
-            "reloadFeatureFlags" -> {
-                reloadFeatureFlags(result)
-            }
-
-            "group" -> {
-                group(call, result)
-            }
-
-            "getFeatureFlag" -> {
-                getFeatureFlag(call, result)
-            }
-
-            "getFeatureFlagPayload" -> {
-                getFeatureFlagPayload(call, result)
-            }
+    private fun configure(call: MethodCall, result: Result) {
+        try {
+            val apiKey: String = call.argument("apiKey") ?: return result.error("Invalid API Key", "API Key is null or empty", null)
+            val host: String = call.argument("host") ?: return result.error("Invalid Host", "Host is null or empty", null)
```

Comment:
@Mastersam07 How about using the default host if the host is not passed?

Reply:
Earlier in the file, there is an `initPlugin` method where we try to initialise the plugin using the values set in the android manifest. When the apiKey is not in the manifest, the initialisation is cancelled.

This leaves the developer to manually call `configure` and pass in the apiKey and host as required parameters.

Reply:
Just read your message again @shuttlers-ifiok . Will modify accordingly.

Reply:
@shuttlers-ifiok this has been treated in #3